### ISSUE TITLE
GPTQ Continuous Batching not working

### ISSUE THREAD

**Issue by bloodsucker99** [huggingface/text-generation-inference#612]
### System Info

I pulled the latest commits few hrs back and built the docker image locally. I tried to use the gptq models such as Bloke 33b with the new changes to TGI regarding gptq. I am able to inference with the model but it seems to only server 1 request at a time. I tried to issue 3 requests from 3 different devices and it waits till one is finished and then continues to the next one.

When GPTQ was initially released in TGI, this was working (sort of but still slow due to how things are for gptq). I tested it with few models on 0.8 version but now it seems to either be a bug or would like to know if its intended.

docker run --gpus all --shm-size 40g -p 1000:80 -v $volume:/data -e GPTQ_BITS=4 -e GPTQ_GROUPSIZE=128 hug:gptq --max-input-length 2048 --max-total-tokens 6000 --model-id TheBloke/WizardLM-33B-V1.0-Uncensored-GPTQ --revision gptq-4bit-128g-actorder_True --num-shard 4 --trust-remote-code --quantize gptq --max-batch-total-tokens 10000

Thank You!

### Information

- [ ] Docker
- [ ] The CLI directly

### Tasks

- [ ] An officially supported command
- [ ] My own modifications

### Reproduction

Running gptq model and inferencing with it on multiple devices at same time

### Expected behavior

Perform continuous batching with gptq models.

**Comment by bloodsucker99** [comment]
nvm it seems to work after i increased the --max-batch-total-tokens. I will do some more testing and close this asap if everything is good

**Comment by Narsil** [comment]
> It seems like if theres a new request, it pauses the other request until its ready to stream 

There are various parameters discoverable in `text-generation-launcher --help` that can manage how we sequence/stack/unstack operations.
What you said is correct, but there's a little more to it.

**Comment by bloodsucker99** [comment]
I tried it but I couldnt get anything to make it answer multiple queries at same time :( Any guidance on this? Like what parameters should i use other than what I have? I have about 90gb vram to work with and would like to use a 33b model. Hopefully once that exllama pr gets merged, it would boost the speeds and quantization will be more usable to serve multi user scenarios :)

**Comment by Narsil** [comment]
It's all your settings.

We are using GPTQ with *many* concurrent users in prod.
But it all boils down to your hardware + settings

**Comment by bloodsucker99** [comment]
@Narsil I got 4 A10G's GPU's. Could you maybe give me like the docker command for smth you would be using to do concurrent requests? I think I should be able to use pretty much same numbers as you would be able to. And from there i can change them but would be nice to have a started point as you are using this in prod. 

Thanks!

**Comment by bloodsucker99** [comment]
Nvm, I figured it out and yes it indeed works. The quality seems to be dropped significantly though but gptq works :)

Still would like to know about WizardCoder support. 

Thanks!

**Comment by Narsil** [comment]
> be dropped significantly though but gptq

Yes it's a model per model thing, possibly linked to seeding and examples too (but it really seems more model per model basis to my very surfacy investigation).

WizardCode is just GPTBigCode, no ? (So it's supported as santacoder models are)

**Comment by bloodsucker99** [comment]
@Narsil Hi! After your quantizing script upgrade, It is not possible to use it anymore. I just get same error for all models i try to quantize. 

weight lm_head.weight does not exist

I also saw that it says loaded model but nothing gets loaded but then realized its loading empty weights. But yea im no longer able to quantize. I tried messing with the code and I also see that its loading the whole thing on 1 gpu so is there way to use all the gpus instead? 

The error im seeing with santacoder models. I neither can load the gptq versions or quantize using the script myself.
